netbox/ipam/forms/model_forms.py

```diff
@@ -351,6 +351,18 @@ def clean(self):
                 'primary_for_parent', "Only IP addresses assigned to an interface can be designated as primary IPs."
             )
 
+        # Do not allow assigning a network ID or broadcast address to an interface.
+        if interface:
+            if address := self.cleaned_data.get('address'):
+                if address.ip == address.network and address.prefixlen not in (31, 32, 127, 128):
```

Comment:
For the sake of completeness, we should split this into separate checks for IPv4 and IPv6 addresses.

Reply:
Done!